msal-dotnet-articles/acquiring-tokens/desktop-mobile/acquiring-tokens-interactively.md

```diff
@@ -1,67 +1,95 @@
 ---
 title: Acquiring tokens interactively
 description: "How to acquire tokens with MSAL.NET and user interaction."
+services: active-directory
+author: Dickson-Mwendia
+manager: CelesteDG
+
+ms.service: active-directory
+ms.subservice: develop
+ms.topic: conceptual
+ms.workload: identity
+ms.date: 12/13/2023
+ms.author: dmwendia
+ms.reviewer: ddelimarsky
+ms.custom: devx-track-csharp, aaddev, devx-track-dotnet
 ---
 
 # Acquiring tokens interactively
 
-The method to use to acquire a token interactively is `IPublicClientApplication.AcquireTokenInteractive`
+Interactive token acquisition requires that the user _interacts_ with an authentication dialog that requests credentials as input and is only available for public client applications. In MSAL.NET, the method to use to acquire a token interactively is <xref:Microsoft.Identity.Client.PublicClientApplication.AcquireTokenInteractive(System.Collections.Generic.IEnumerable{System.String})>.
 
-The following example shows minimal code to get a token for reading the user's profile with Microsoft Graph.
+The following example shows barebones code to get a token for reading the user's profile with Microsoft Graph:
 
 ```csharp
-string[] scopes = new string[] {
-  "user.read"
-};
-var app = PublicClientApplicationBuilder.Create(clientId).Build();
+string[] scopes = new string[] { "user.read" };
+
+var app = PublicClientApplicationBuilder.Create("YOUR_CLIENT_ID")
+    .WithDefaultRedirectUri()
+    .Build();
+
 var accounts = await app.GetAccountsAsync();
+
 AuthenticationResult result;
-try {
-  result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
-    .ExecuteAsync();
-} catch (MsalUiRequiredException) {
-  result = await app.AcquireTokenInteractive(scopes)
-    .ExecuteAsync();
+try
+{
+    result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
+      .ExecuteAsync();
+}
+catch (MsalUiRequiredException)
+{
+    result = await app.AcquireTokenInteractive(scopes).ExecuteAsync();
 }
 ```
 
-## Mandatory parameters
+>[!NOTE]
+>To use <xref:Microsoft.Identity.Client.ClientApplicationBase.AcquireTokenSilent(System.Collections.Generic.IEnumerable{System.String},Microsoft.Identity.Client.IAccount)> the developer needs to set up a token cache. Without a token cache, the interactive prompt will always be shown, even if the user has previously logged in. To learn more about setting up a token cache, refer to [Token cache serialization in MSAL.NET](../../how-to/token-cache-serialization.md).
```

Comment:
Good note. I updated the AcquireTokenX method comments with the same comment. Good to mention this in a few places.